sklearn/inspection/partial_dependence.py

```diff
@@ -345,22 +352,30 @@ def partial_dependence(estimator, X, features, response_method='auto',
                 "With the 'recursion' method, the response_method must be "
                 "'decision_function'. Got {}.".format(response_method)
             )
+        # msg = "'estimator' parameter must be a fitted estimator"
+        # if isinstance(estimator, BaseGradientBoosting):
+        #     fitted_attribute = 'estimators_'
+        # else:
+        #     fitted_attribute = 'n_iter_'
+        # check_is_fitted(estimator, fitted_attribute, msg=msg)
```

Comment:
@NicolasHug I found the reason why was I was doing so. Using an `HistGradientBoostingResgressor` unfitted will lead to:

```pydb
    def _compute_partial_dependence_recursion(self, grid, target_features):
        """Fast partial dependence computation.
    
        Parameters
        ----------
        grid : ndarray, shape (n_samples, n_target_features)
            The grid points on which the partial dependence should be
            evaluated.
        target_features : ndarray, shape (n_target_features)
            The set of target features for which the partial dependence
            should be evaluated.
    
        Returns
        -------
        averaged_predictions : ndarray, shape \
                (n_trees_per_iteration, n_samples)
            The value of the partial dependence function on each grid point.
        """
        grid = np.asarray(grid, dtype=X_DTYPE, order='C')
        averaged_predictions = np.zeros(
>           (self.n_trees_per_iteration_, grid.shape[0]), dtype=Y_DTYPE)
E       AttributeError: 'HistGradientBoostingRegressor' object has no attribute 'n_trees_per_iteration_'
```

Probably, we should make an earlier check at the beginning of the function to check that the estimator is fitted to fail always as early as possible. Here, the difficulties is that classifier will always have an argument `n_classes_` while this is not true for the regressor. I would need to make check for those.